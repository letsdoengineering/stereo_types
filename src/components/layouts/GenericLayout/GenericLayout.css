.generic-layout_full-page {
    text-align: center;
    margin: 16px;
}

/* Extra small devices (phones, 600px and down) */
@media only screen and (max-width: 600px) {
    .generic-layout_full-page {
        margin: 0;
    }
}

.generic-layout_main-content {
    margin: 0 auto;
    min-height: 400px;
    max-width: 1300px;
    min-width: 450px;
}
