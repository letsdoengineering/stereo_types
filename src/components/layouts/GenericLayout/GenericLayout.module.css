.main-container {
    margin: 0 auto;
    min-height: 40rem;
    max-width: 78rem;
    min-width: 60rem;
}

.full-page {
    text-align: center;
    margin: 0 1rem;
}