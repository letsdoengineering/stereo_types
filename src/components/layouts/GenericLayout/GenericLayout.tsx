import React from 'react'
import Head from '../../basics/Head/Head'
import Heading from '../../basics/Heading/Heading'
import * as genericLayoutStyles from './GenericLayout.module.css'

/** GenericLayout: generic layout with a footer and header used to e.g. wrap other layouts */
const GenericLayout: React.FC = ({ children }) => (
  <div className={genericLayoutStyles.fullPage}>
    <Head title='Engineering Survey for wee ones' />
    <main>
      <Heading>Engineering Survey for wee ones</Heading>
      <div className={genericLayoutStyles.mainContainer}>{children}</div>
    </main>
  </div>
)

export default GenericLayout
