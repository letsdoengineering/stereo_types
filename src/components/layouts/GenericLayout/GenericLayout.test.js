import * as React from 'react'
import GenericLayout from './GenericLayout'
import renderer from 'react-test-renderer'
import { render } from '@testing-library/react'
import { axe } from 'jest-axe'

const mockContent = {
  footer: 'some footer',
}

describe('GenericLayout', () => {
  it('has not changed the JSX layout of the page from previous snapshot', () => {
    const tree = renderer.create(<GenericLayout content={mockContent} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders the page DOM that is accessible', async () => {
    const { container } = render(<GenericLayout content={mockContent} />)
    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
