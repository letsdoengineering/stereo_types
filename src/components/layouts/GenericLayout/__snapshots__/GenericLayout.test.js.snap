// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenericLayout has not changed the JSX layout of the page from previous snapshot 1`] = `
<div
  className="fullPage"
>
  <main>
    <h1
      className="title title1 weightNormal"
    >
      Engineering Survey for wee ones
    </h1>
    <div
      className="mainContent"
    />
  </main>
</div>
`;
